source/assets/js/playground/console-utils.ts

```diff
@@ -36,115 +36,112 @@ export type ConsoleLog = ConsoleLogDebug | ConsoleLogWarning | ConsoleLogError;
  * Write with `innerText` and then retrieve using `innerHTML` to encode message
  * for safe display.
  * @param  {string} message The user-submitted string
- * @param  {string} [safeLink] A known safe URL. If `safeLink` is present in
- * `message`, it will be wrapped in an anchor element.
  * @return {string} The sanitized string
  */
-function encodeHTML(message: string, safeLink?: string): string {
+function encodeHTML(message: string): string {
   const el = document.createElement('div');
   el.innerText = message;
-  let html = el.innerHTML;
-  if (safeLink) {
-    html = html.replace(
-      safeLink,
-      `<a href="${safeLink}" target="_blank">${safeLink}</a>`
-    );
-  }
-  return html;
+  return el.innerHTML;
 }
 
+// Converts 0-indexed number to 1-indexed string, or empty string if undefined.
 function lineNumberFormatter(number?: number): string {
   if (number === undefined) return '';
   number = number + 1;
   return `${number}`;
 }
 
+// Returns undefined if no range, or a link to the state, including range.
 function selectionLink(
   playgroundState: PlaygroundState,
-  range?: PlaygroundState['selection']
-): string {
-  if (!range) return '';
+  range: PlaygroundSelection
+): string | undefined {
+  if (!range) return undefined;
   return serializeState({...playgroundState, selection: range});
 }
 
+// Returns a safe HTML string for a console item.
 export function displayForConsoleLog(
   item: ConsoleLog,
   playgroundState: PlaygroundState
 ): string {
-  const data: {
-    type: string;
-    lineNumber?: number;
-    message: string;
-    safeLink?: string;
-    range?: PlaygroundState['selection'];
-  } = {
-    type: item.type,
-    lineNumber: undefined,
-    message: '',
-    safeLink: undefined,
-    range: null,
-  };
+  const type = item.type;
+  let lineNumber = undefined;
+  let message = '';
+  let safeLink = undefined;
+  let range: PlaygroundSelection = null;
+
   if (item.type === 'error') {
     if (item.error instanceof Exception) {
       const span = item.error.span;
-      data.lineNumber = span.start.line;
-      data.range = [
+      lineNumber = span.start.line;
+      range = [
         span.start.line + 1,
         span.start.column + 1,
         span.end.line + 1,
         span.end.column + 1,
       ];
     }
-    data.message = item.error?.toString() || '';
+    message = item.error?.toString() || '';
   } else if (['debug', 'warn'].includes(item.type)) {
-    data.message = item.message;
-    let lineNumber = item.options.span?.start?.line;
+    message = item.message;
+    let _lineNumber = item.options.span?.start?.line;
```

Comment:
I'd like to clean up this logic and get rid of the confusing `lineNumber`/`_lineNumber` distinction.

`SourceSpan.start` is always defined, so `_lineNumber` will by undefined if and only if `item.options.span` is undefined. That means that instead of `if (typeof _lineNumber === 'undefined')` you could just make that an `else` clause on `if (item.options.span)`. At which point you can just assign `lineNumber` directly rather than using an intermediate variable for it.